src/univariate/continuous/normal.jl

```diff
@@ -87,7 +89,7 @@ cf(d::Normal, t::Real) = exp(im * t * d.μ - d.σ^2/2 * t^2)
 
 #### Sampling
 
-rand(rng::AbstractRNG, d::Normal) = d.μ + d.σ * randn(rng)
+rand(rng::AbstractRNG, d::Normal{T}) where {T} = d.μ + d.σ * randn(rng, T)
```

Comment:
Not sure if we shouldn't watch out here, since `randn` does not support all types

Reply:
Precisely the concern I mentioned above; although on balance, I think this is an improvement. The prior behavior wasn't technically correct, either due to the wrong output type, or the right output type but the code not actually doing what one would expect (as shown in the examples I gave above).